### ISSUE TITLE
Missing labels changes colors in the heatmaps 'Following `at` are removed:'

### ISSUE THREAD

**Issue by nroak** [jokergoo/ComplexHeatmap#998]
I'm running ComplexHeatmap/Oncoprint using alter_fun and col parameters. But I'm getting a message for missing levels, which is fine. But when it plots the heatmap, it's switching colors from the ones I have defined.

Here's my full code function-
```
col <- c("splicing" = "#E69F00", "nonsynonymous_SNV" = "#56B4E9", "stopgain" = "#009E73", "stoploss" = "#000000", "startloss" = "#F0E442","frameshift_insertion" = "#e37a54", "frameshift_deletion" = "#D55E00", "CHT" = "#CC79A7", "Duplication" = "#1AFF1A", "Deletion" = "#4B0092")
alter_fun = list(
  background = function(x, y, w, h) {
    grid.rect(x, y, w-unit(1, "pt"), h-unit(1, "pt"),
              gp = gpar(fill = "#dce0df", col = NA,lwd=2))
  },
  Duplication = function(x, y, w, h) {
    grid.rect(x, y, w-unit(1, "pt"), h-unit(1, "pt"),
              gp = gpar(fill = col["Duplication"], col = NA))
  },
  Deletion = function(x, y, w, h) {
    grid.rect(x, y, w-unit(1, "pt"), h-unit(1, "pt"),
              gp = gpar(fill = col["Deletion"], col = NA))
  },
  splicing = function(x, y, w, h) {
    grid.rect(x, y, w-unit(1, "pt"), h-unit(1, "pt"),
              gp = gpar(fill = col["splicing"], col = NA))
  },
  nonsynonymous_SNV = function(x, y, w, h) {
    grid.rect(x, y, w-unit(1, "pt"), h-unit(1, "pt"),
              gp = gpar(fill = col["nonsynonymous_SNV"], col = NA))
  },
  stopgain = function(x, y, w, h) {
    grid.rect(x, y, w-unit(1, "pt"), h-unit(1, "pt"),
              gp = gpar(fill = col["stopgain"], col = NA))
  },
  stoploss = function(x, y, w, h) {
    grid.rect(x, y, w-unit(1, "pt"), h-unit(1, "pt"),
              gp = gpar(fill = col["stoploss"], col = NA))
  },
  startloss = function(x, y, w, h) {
    grid.rect(x, y, w-unit(1, "pt"), h-unit(1, "pt"),
              gp = gpar(fill = col["startloss"], col = NA))
  },
  frameshift_insertion = function(x, y, w, h) {
    grid.rect(x, y, w-unit(1, "pt"), h-unit(1, "pt"),
              gp = gpar(fill = col["frameshift_insertion"], col = NA))
  },
  frameshift_deletion = function(x, y, w, h) {
    grid.rect(x, y, w-unit(1, "pt"), h-unit(1, "pt"),
              gp = gpar(fill = col["frameshift_deletion"], col = NA))
  },
  CHT = function(x, y, w, h) {
    grid.rect(x, y, w-unit(1, "pt"), h-unit(1, "pt"),
              gp = gpar(fill = col["CHT"], col = NA))
  }
)
plot_heatmap <- function(newDF){
  newDF[is.na(newDF)] = ""
  split<-data.frame()
  split<- data.frame(gene=rownames(newDF))

  heatmap_legend_param = list(title = "Variant Type", at = c("nonsynonymous_SNV", "splicing", "stopgain", "stoploss", "startloss","frameshift_insertion", "frameshift_deletion", "CHT", "Duplication", "Deletion"),
                              labels = c("nonsynonymous_SNV", "splicing", "stopgain", "stoploss", "startloss","frameshift_insertion", "frameshift_deletion", "CHT", "Duplication", "Deletion"),
                              title_gp = gpar(fontsize = 11,fontface="bold"), labels_gp = gpar(fontsize = 10), ncol = 1 )
  sample_info <- sample_info[match(colnames(newDF), sample_info$sample),]
  
  ht <- oncoPrint(as.matrix(newDF),
                  alter_fun = alter_fun, col = col, show_pct = FALSE, show_column_names = TRUE, 
                  column_names_gp = gpar(fontsize = 12), row_names_gp = gpar(fontsize = 12),  row_names_side = "left",
                  row_split = split,row_gap=unit( 3, "mm"), border = TRUE, row_title_rot = 0,row_title_gp = gpar(fontface="bold"),column_order= colnames(newDF),
                  column_title = column_title, heatmap_legend_param = heatmap_legend_param)
  return(ht)
}
```

I defined the non_synonymous to be lightblue, stopgain as green, and Deletion to be purple, but they all appear to be switched by one. 
<img width="449" alt="Screen Shot 2022-10-12 at 1 02 59 PM" src="https://user-images.githubusercontent.com/24547900/195415898-6da0a49a-8975-4083-a56d-f485b73712d6.png">
<img width="144" alt="Screen Shot 2022-10-12 at 1 02 51 PM" src="https://user-images.githubusercontent.com/24547900/195415916-9b3fc4e0-fa6c-406b-8e22-32d885c410a1.png">

**Comment by nroak** [comment]
The new fix has worked! Thank you for your quick fix.
